spec/services/untrack_service_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe UntrackService do
+  let (:repository) { create(:repository) }
+  let!(:hook) { create(:hook, repository_id: repository.id, active: true) }
+  let(:pull_request) { create(:pull_request, repository_id: repository.id) }
+  let(:pull_request_relation) { create(pull_request_relation, pull_request_id: pull_request.id) }
+
+  describe "#destroy_entities" do
+    def build
+      described_class.new(repo: repository)
+    end
+
+    it 'call methods' do
+      expect_any_instance_of(described_class).to receive(:destroy_hook)
+      expect_any_instance_of(described_class).to receive(:destroy_pull_request)
+      build.destroy_entities
+    end
+
+    it 'deletes hook' do
+      build.destroy_entities
+      expect(Hook.where(repository_id: repository.id)).to be_empty
```

Comment:
Acá también eliminaría las consulstas y cambiaría por `repo.hook to be_nil`

Reply:
Pero no es un solo elemento, son varios y estamos revisando uno en particular. Queremos que se borre el asociado, no todos. Entonces lo que realmente sirve hacer (si no quieres que use el modelo) es
```
expect(repository.hooks.where(id: hook.id)).to be_empty
```

Reply:
Si, algo así. Si son varios, puede ser que previos al test hayan más hooks asociados y cuando borres el que querías borrar, `Hook.where(repository_id: repository.id)` no sea vacío